plugin/storage/es/mappings/mapping.go

```diff
@@ -9,36 +9,66 @@ import (
 	"strings"
 
 	"github.com/jaegertracing/jaeger/pkg/es"
+	"github.com/jaegertracing/jaeger/pkg/es/config"
 )
 
 // MAPPINGS contains embedded index templates.
 //
 //go:embed *.json
 var MAPPINGS embed.FS
 
-// MappingBuilder holds parameters required to render an elasticsearch index template
+// MappingBuilder holds common parameters required to render an elasticsearch index template
 type MappingBuilder struct {
-	TemplateBuilder              es.TemplateBuilder
-	Shards                       int64
-	Replicas                     int64
-	PrioritySpanTemplate         int64
-	PriorityServiceTemplate      int64
-	PriorityDependenciesTemplate int64
-	PrioritySamplingTemplate     int64
-	EsVersion                    uint
-	IndexPrefix                  string
-	UseILM                       bool
-	ILMPolicyName                string
+	TemplateBuilder es.TemplateBuilder
+	Indices         config.Indices
+	EsVersion       uint
+	IndexPrefix     string
+	UseILM          bool
+	ILMPolicyName   string
+}
+
+// IndexTemplateOptions separate holds parameters required to render an elasticsearch index template
+type IndexTemplateOptions struct {
+	IndexPrefix   string
+	UseILM        bool
+	ILMPolicyName string
+	config.IndexOptions
+}
+
+func (mb *MappingBuilder) getMappingTemplateOptions(mapping string) *IndexTemplateOptions {
+	mappingOpts := &IndexTemplateOptions{}
+	mappingOpts.IndexPrefix = mb.IndexPrefix
+	mappingOpts.UseILM = mb.UseILM
+	mappingOpts.ILMPolicyName = mb.ILMPolicyName
+
+	if strings.Contains(mapping, "span") {
```

Comment:
these if statements are meant to be mutually exclusive, which is not evident from the code. Maybe switch statement instead?
```
switch {
case strings.Contains(mapping, "span"):
    mappingOpts.IndexOptions = mb.Indices.Spans
case strings.Contains(mapping, "service"):
    mappingOpts.IndexOptions = mb.Indices.Services
case strings.Contains(mapping, "dependencies"):
    mappingOpts.IndexOptions = mb.Indices.Dependencies
case strings.Contains(mapping, "sampling"):
    mappingOpts.IndexOptions = mb.Indices.Sampling
}
```